java/performance/pom.xml

```diff
@@ -22,7 +22,9 @@
   <description>JMH Performance benchmarks for other Arrow libraries.</description>
 
   <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```

Comment:
Here in the previous PR the following component has been removed, but we haven't reverted it back here.
```pom
<plugin>
  <groupId>org.apache.maven.plugins</groupId>
  <artifactId>maven-compiler-plugin</artifactId>
  <configuration combine.self="override">
    <compilerVersion>${javac.target}</compilerVersion>
    <source>${javac.target}</source>
    <target>${javac.target}</target>
  </configuration>
</plugin>
```
And in the main we have 

```pom
<plugin>
  <groupId>org.apache.maven.plugins</groupId>
  <artifactId>maven-compiler-plugin</artifactId>
  <configuration>
    <annotationProcessorPaths combine.children="append">
      <path>
        <groupId>org.openjdk.jmh</groupId>
        <artifactId>jmh-generator-annprocess</artifactId>
        <version>${jmh.version}</version>
      </path>
    </annotationProcessorPaths>
  </configuration>
</plugin>
```

So I hope this is okay?

Reply:
I'm not particularly bothered so long as CI passes.